db/active_replicator_checkpointer.go

```diff
@@ -329,62 +355,87 @@ func (c *Checkpointer) fetchCheckpoints() error {
 	base.DebugfCtx(c.ctx, base.KeyReplicate, "got remote checkpoint: %v", remoteCheckpoint)
 	c.lastRemoteCheckpointRevID = remoteCheckpoint.Rev
 
-	// If checkpoint hash has changed, reset checkpoint to zero.  Shouldn't need to persist the updated checkpoints in
-	// the case both get rolled back to zero, can wait for next persistence
-	if localCheckpoint.ConfigHash != c.configHash {
-		localCheckpoint.LastSeq = ""
-	}
-
-	if remoteCheckpoint.ConfigHash != c.configHash {
-		remoteCheckpoint.LastSeq = ""
-	}
-
-	// If localSeq and remoteSeq match, we'll use this value.
 	localSeq := localCheckpoint.LastSeq
 	remoteSeq := remoteCheckpoint.LastSeq
+
+	// If localSeq and remoteSeq match, we'll use localSeq as the value.
 	checkpointSeq := localSeq
 
-	// Determine the lowest sequence if they don't match
-	if localSeq != remoteSeq {
-		base.DebugfCtx(c.ctx, base.KeyReplicate, "sequences mismatched, finding lowest of %q %q", localSeq, remoteSeq)
-		localSeqVal, err := parseIntegerSequenceID(localSeq)
-		if err != nil {
-			return err
+	// If both local and remote SGR2 checkpoints are missing, fetch SGR1 checkpoint as fallback as a one-time upgrade.
+	// Explicit checkpoint resets are expected to only reset one side of the checkpoints, so shouldn't fall into this handling.
+	var usedSGR1Checkpoint bool
+	if localSeq == "" && remoteSeq == "" && c.sgr1CheckpointID != "" {
+		base.Infof(base.KeyReplicate, "Attempting to fetch SGR1 checkpoint as fallback: %v", c.sgr1CheckpointID)
```

Comment:
Sticking lines 368-382 into a c.getSGR1Checkpoint() function would mirror the pattern used for getLocalCheckpoint()/getRemoteCheckpoint() and reduce the code complexity a bit here.